src/FastSerialization/SegmentedDictionary/SegmentedDictionary.cs

```diff
@@ -781,7 +781,8 @@ public int EnsureCapacity(int capacity)
                 throw new ArgumentOutOfRangeException(nameof(capacity));
             }
 
-            var currentCapacity = _entries.Capacity;
+            // Normal usage of a dictionary should never ask for a capacity that exceeds int32.MaxValue.
```

Comment:
Is the cast here necessary?

Reply:
I believe so, since the underlying data structure supports long indices, but the dictionary does not.

Reply:
Wondering if the cast should actually be on capacity and a cast to (long) so that this doesn't truncate the value if it's greater than int.MaxValue?

Reply:
the issue is the return value, more than that the value of the capacity arg?

Reply:
Oh, I see.  In that case, should not matter because the return value has to be an int anyway.  Got it.

Reply:
correct. Are you ok in resolving it then?